front-end/src/views/Meal.jsx

```diff
@@ -0,0 +1,87 @@
+import './Meal.css'
+import { useState, useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+import { baseUrl } from '../apiurl'
+import { BeatLoader } from 'react-spinners'
+
+export function Meal() {
+  const { mealId } = useParams()
+  const [mealData, setMealData] = useState(undefined)
+
+  const fetchMealDetails = async () => {
+    if (mealData) return
+
+    try {
+      const response = await fetch(`${baseUrl()}meals/${mealId}`)
+      if (response.ok) {
+        const data = await response.json()
+
+        const reservationResponse = await fetch(
+          `${baseUrl()}meals/${mealId}/reservations`
+        )
+        const reservation = await reservationResponse.json()
+
+        const date = new Date(data.meal_time)
+        setMealData({
+          ...data,
+          mealTime: `${date.getDate()}-${
+            date.getMonth() + 1
+          }-${date.getFullYear()} ${date.getHours()}:${date.getMinutes()}`,
+          availableMeals: reservation.availableMeals,
+          maxReservation: reservation.maxReserve,
+        })
+      } else {
+        throw new Error('Failed to fetch meal details')
```

Comment:
right now you do not have any error boundaries, this would crash the app. You could just show an error message to the user instead of shutting down the app